### PR TITLE
feat: Port over calcite-app-components

### PR DESCRIPTION
**Pull request by driskull**
Port over calcite-app-components #687

## Notes

- App components are currently excluded from the build and storybook.

## Todo

- Setup/modify teams channel notifications (@driskull) #786
- Get release scripts working (@jcfranco ) – will be tackled separately by #688 
- Fix sizing styles (@asangma) – will be tackled separately by #687 
- Merge variables and organize styles (@asangma) – will be tackled separately by #687 
- Improved linting will be ported separately (#688)

### CODE REVIEW

**Comment by driskull** [comment]
Got the storybooks working together. I think this is ready for more eyes.

File: .github/ISSUE_TEMPLATE/accessibility.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (paulcpederson) [inline .github/ISSUE_TEMPLATE/accessibility.md:1]: Sweet, this issue templates will be really nice to have! 👍

Comment (driskull) [reply .github/ISSUE_TEMPLATE/accessibility.md:1]: yeah> I copied EVERYTHING over pretty much so some things will need to be tweaked or reworded to encompass both types of components. I'll continue to work on that stuff but the Storybook is working for testing things out.

File: .storybook/main.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  addons: [
+    "@storybook/addon-backgrounds/register",
+    "@storybook/addon-knobs/register",
+    "@storybook/addon-notes/register-panel",
+    "@storybook/addon-a11y/register"
+  ],
+  presets: ["@storybook/addon-docs/preset"],
+  stories: ["../src/**/*.stories.(mdx|ts|js)"]
```
Comment (paulcpederson) [inline .storybook/main.ts:9]: oh nice, mdx support will be great

File: .storybook/utils.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * This transforms a component markdown to properly render in Storybook notes.
+ */
+export const parseReadme = (content: string): string => {
+  return (
+    content
+      // markdown uses relative paths for component links
+      .replace(
+        /\.\.\//g,
+        "https://github.com/Esri/calcite-app-components/tree/master/src/components/"
```
Comment (paulcpederson) [inline .storybook/utils.ts:36]: This should probably be the calcite-components url now, yeah?

File: .storybook/main.ts (hunk 2 above)
Comment (driskull) [reply .storybook/main.ts:9]: Yes, we're using these to have some documentation generated from the markdown files.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Code of Conduct
```
Comment (paulcpederson) [inline CODE_OF_CONDUCT.md:1]: 😄 ✨

File: PROJECT_GUIDELINES.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
+# Project Guidelines
+
+Hey there and thanks for stopping by! We welcome all feedback and contributions to [Calcite App Components](https://github.com/Esri/calcite-components/blob/master/README.md). If you are interested in getting involved, we ask that you give our guidelines a read. Below is a list of things we've found work well for us as we continue developing our components.
```
Comment (paulcpederson) [inline PROJECT_GUIDELINES.md:3]:
```suggestion
Hey there and thanks for stopping by! We welcome all feedback and contributions to [Calcite Components](https://github.com/Esri/calcite-components/blob/master/README.md). If you are interested in getting involved, we ask that you give our guidelines a read. Below is a list of things we've found work well for us as we continue developing our components.
```

File: src/scss/placeholders/_host.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
+%component-host {
+  /* Base ":host" styles for the component */
+  box-sizing: border-box;
+  * {
+    box-sizing: border-box;
+  }
+  color: var(--calcite-app-foreground);
+  font-family: var(--calcite-app-font-family);
```
Comment (paulcpederson) [inline src/scss/placeholders/_host.scss:8]: adding the full font stack to every component seems like a waste. Could just use `inherit`

Comment (driskull) [reply src/scss/placeholders/_host.scss:8]: Yeah, I think we can work with @asangma to clean these up and get them in sync with `calcite-components`

**Comment by paulcpederson** [comment]
Re:merging scss vars/styles, it may make sense to use this as an opportunity to start moving to the new tailwind approach. We should probably merge https://github.com/Esri/calcite-components/pull/702 into the `master` branch to make these available, and then we could just move the app-components over to using `@apply` as I think this is where we're headed anyways with the whole library.

**Pull request merged**